internal/circle/client.go

```diff
@@ -40,29 +41,31 @@ type ClientInterface interface {
 
 // Client provides methods to interact with the Circle API.
 type Client struct {
-	BasePath      string
-	APIKey        string
-	httpClient    httpclient.HttpClientInterface
-	tenantManager tenant.ManagerInterface
+	BasePath       string
+	APIKey         string
+	httpClient     httpclient.HttpClientInterface
+	tenantManager  tenant.ManagerInterface
+	monitorService monitor.MonitorServiceInterface
 }
 
 // ClientFactory is a function that creates a ClientInterface.
-type ClientFactory func(networkType utils.NetworkType, apiKey string, tntManager tenant.ManagerInterface) ClientInterface
+type ClientFactory func(networkType utils.NetworkType, apiKey string, tntManager tenant.ManagerInterface, monitorSvc monitor.MonitorServiceInterface) ClientInterface
```

Comment:
We should consider using a struct as the input variable for this method. Maybe for the next change.

Reply:
addressed in a712a9b61f6dafc8344674f3adc55c3525d11617